examples/k-means/k-means.hpp

```diff
@@ -172,26 +172,26 @@ class KMeansModel
 //! input and an output parameter. The method returns a std::pair<Point2D,
 //! size_t> = Point2DClusterId into the centroids for each input point.
 template <typename Point, typename InStack>
-auto KMeans(const DIA<Point, InStack>& input_points,
-            size_t dimensions, size_t num_clusters, size_t iterations) {
+auto KMeans(const DIA<Point, InStack>& input_points, size_t dimensions,
+            size_t num_clusters, size_t iterations, double epsilon) {
 
     auto points = input_points.Cache();
 
+	bool break_condition = false;
+
     using ClosestCentroid = ClosestCentroid<Point>;
     using CentroidAccumulated = CentroidAccumulated<Point>;
 
-    DIA<Point> centroids = points.Keep().Sample(num_clusters);
+	std::vector<Point> local_centroids = 
+		points.Keep().Sample(num_clusters).AllGather();
 
-    for (size_t iter = 0; iter < iterations; ++iter) {
+    for (size_t iter = 0; iter < iterations && !break_condition; ++iter) {
 
-        // handling this local variable is difficult: it is calculated as an
-        // Action here, but must exist later when the Map() is
-        // processed. Hhence, we move it into the closure.
-        std::vector<Point> local_centroids = centroids.AllGather();
+		std::vector<Point> old_local_centroids(local_centroids);
 
         // calculate the closest centroid for each point
         auto closest = points.Keep().Map(
-            [local_centroids = std::move(local_centroids)](const Point& p) {
+            [local_centroids](const Point& p) {
```

Comment:
why copy here? shouldn't this be a ref-capture?

Reply:
yes, ref capture works too.